dbms/src/Interpreters/Aggregator.h

```diff
@@ -1015,25 +1024,10 @@ class Aggregator
     void setCancellationHook(CancellationHook cancellation_hook);
 
     /// For external aggregation.
-    void writeToTemporaryFile(AggregatedDataVariants & data_variants, const FileProviderPtr & file_provider);
-
-    bool hasTemporaryFiles() const { return !temporary_files.empty(); }
-
-    struct TemporaryFiles
-    {
-        std::vector<std::unique_ptr<Poco::TemporaryFile>> files;
-        size_t sum_size_uncompressed = 0;
-        size_t sum_size_compressed = 0;
-        mutable std::mutex mutex;
-
-        bool empty() const
-        {
-            std::lock_guard lock(mutex);
-            return files.empty();
-        }
-    };
-
-    const TemporaryFiles & getTemporaryFiles() const { return temporary_files; }
+    void spill(AggregatedDataVariants & data_variants);
+    void finishSpill();
+    BlockInputStreams restoreSpilledData();
+    bool hasSpilledData() const { return spiller != nullptr && spiller->hasSpilledData(); }
```

Comment:
Is the `spiller->hasSpilledData()` here correct?
Because the way AggregateBlockInputStream is used is
```
if (aggregator->hasSpilledData())
    aggregator->spill();
```
But before `aggregator->spill`, `aggregator->hasSpilledData()` is always false.
Is my understanding correct?

Reply:
Yes, it is in the merge stage, in agg stage, it does not check `spiller->hasSpilledData()`